### PR TITLE
APS-404 AP Area Backfill Fixes

### PR DESCRIPTION
**Pull request by davidatkinsuk**

### CODE REVIEW

File: src/test/kotlin/uk/gov/justice/digital/hmpps/approvedpremisesapi/unit/service/cas1/Cas1UserMappingServiceTest.kt [hunk 1]
```diff
@@ -176,7 +182,7 @@ class Cas1UserMappingServiceTest {
     @Test
     fun `throws error with details when resolved area id doesnt exist in ap area table`() {
       val staffUserDetails = StaffUserDetailsFactory()
-        .withProbationAreaCode("XX")
+        .withProbationAreaCode("XXX")
```
Comment (edavey) [inline src/test/kotlin/uk/gov/justice/digital/hmpps/approvedpremisesapi/unit/service/cas1/Cas1UserMappingServiceTest.kt:185]: I should have spotted this earlier!

File: src/main/kotlin/uk/gov/justice/digital/hmpps/approvedpremisesapi/service/UserService.kt [hunk 2]
```diff
@@ -199,26 +199,34 @@ class UserService(
...
-      }
+      user = updateUser(user, deliusUser, forService)
+    }
 
-      if (forService == ServiceName.approvedPremises) {
-        user.apArea = cas1UserMappingService.determineApArea(user.probationRegion, deliusUser)
+    return AuthorisableActionResult.Success(user)
+  }
+
+  fun updateUser(
```
Comment (edavey) [inline src/main/kotlin/uk/gov/justice/digital/hmpps/approvedpremisesapi/service/UserService.kt:208]: that's a neat extraction

**Review by edavey — APPROVED** [review]
👍

File: src/test/kotlin/uk/gov/justice/digital/hmpps/approvedpremisesapi/unit/service/cas1/Cas1UserMappingServiceTest.kt (hunk 1 above)
Comment (davidatkinsuk) [reply src/test/kotlin/uk/gov/justice/digital/hmpps/approvedpremisesapi/unit/service/cas1/Cas1UserMappingServiceTest.kt:185]: Easily done for both of us!

**Pull request merged**